Source/JavaScriptCore/heap/WeakSetInlines.h

```diff
@@ -41,6 +41,7 @@ inline WeakImpl* WeakSet::allocate(JSValue jsValue, WeakHandleOwner* weakHandleO
     weakSet.m_allocator = allocator->next;
 
     WeakImpl* weakImpl = WeakBlock::asWeakImpl(allocator);
+    container.vm().heap.didAllocate(sizeof(WeakImpl));
```

Comment:
Maybe didAllocateInBlock needs to be called?

https://github.com/WebKit/WebKit/blob/f1f86afb2e78feb7393d53dd8fdfd901e06038c3/Source/JavaScriptCore/heap/LocalAllocator.cpp#L255

Reply:
Maybe I am reading this wrong. My understanding is that didAllocateInBlock is used to make sure that weak sets get updated. That is, if I allocate Foo, then didAllocateInBlock finds any WeakSets containing Foo, and clears them. This is allocating the WeakSet itself, so we don't need didAllocateInBlock. didAllocate is just for tracking the number of newly allocated bytes, but since this file is inlinable, it can be included from other compilation units, and thus we need to export it. Let's see what a second round of EWS thinks, to see if the failures were a random fluke or caused by this patch.